src/ssl.c

```diff
@@ -13603,42 +14123,92 @@ int wolfSSL_X509_LOOKUP_add_dir(WOLFSSL_X509_LOOKUP* lookup, const char* dir,
 
 
 int wolfSSL_X509_LOOKUP_load_file(WOLFSSL_X509_LOOKUP* lookup,
-                                 const char* file, long len)
+                                 const char* file, long type)
 {
+#ifndef NO_FILESYSTEM
+    int           ret = 0;
+    XFILE         fp;
+    long          sz;
+    byte*         pem;
+    WOLFSSL_X509* x509;
+
+    if (type != X509_FILETYPE_PEM)
+        return BAD_FUNC_ARG;
+
+    fp = XFOPEN(file, "r");
+    if (fp == NULL)
+        return BAD_FUNC_ARG;
+
+    XFSEEK(fp, 0, XSEEK_END);
+    sz = XFTELL(fp);
+    XREWIND(fp);
+
+    pem = (byte*)XMALLOC(sz, 0, DYNAMIC_TYPE_TMP_BUFFER);
+    if (pem == NULL)
+        return MEMORY_ERROR;
+
+    /* Read in file which may be a CRL or certificate. */
+    if (XFREAD(pem, sz, 1, fp) != 1)
+        goto end;
+
+    if (XSTRNSTR((char*)pem, BEGIN_X509_CRL, sz) != NULL) {
```

Comment:
I assume the check is checking the result of the XFTELL? Added check.
Added cast in XFREAD call.